panther_analysis_tool/main.py

```diff
@@ -347,7 +353,10 @@ def upload_zip(backend: BackendClient, args: argparse.Namespace, archive: str) -
 
         while True:
             try:
-                response = backend.bulk_upload(upload_params)
+                if use_async:
+                    response = backend.async_bulk_upload(upload_params)
+                else:
+                    response = backend.bulk_upload(upload_params)
```

Comment:
I might have missed this somewhere in the PR, but do we have logic that says we should only do async upload if we are using the public api?

Reply:
yup that case is handled. I just added a supports_async_bulk_upload method on the various backend clients .. the lambda client will default to no. and the graphql client will be true if the instance supports it.